rust/theoros/src/services/hyperlane/mod.rs

```diff
@@ -0,0 +1,54 @@
+use pragma_utils::services::Service;
+use starknet::core::types::Felt;
+use tokio::task::JoinSet;
+
+use crate::{rpc::HyperlaneCalls, AppState};
+
+#[derive(Clone)]
+pub struct HyperlaneService {
+    state: AppState,
+    merkle_tree_hook_address: Felt,
+}
+
+#[async_trait::async_trait]
+impl Service for HyperlaneService {
+    async fn start(&mut self, join_set: &mut JoinSet<anyhow::Result<()>>) -> anyhow::Result<()> {
+        let service = self.clone();
+        join_set.spawn(async move {
+            tracing::info!("🧩 Hyperlane service started");
+            service.run_forever().await?;
+            Ok(())
+        });
+        Ok(())
+    }
+}
+
+impl HyperlaneService {
+    pub fn new(state: AppState, merkle_tree_hook_address: Felt) -> Self {
+        Self { state, merkle_tree_hook_address }
+    }
+
+    pub async fn run_forever(&self) -> anyhow::Result<()> {
+        loop {
+            let storage = self.state.storage.validators().all().await;
+            for (validator, checkpoint) in storage {
+                tracing::info!("Validator: {:?} - Storage: {:?}", validator, checkpoint);
+                let index = self.get_latest_index().await?;
+                let fetcher = checkpoint.build().await?;
```

Comment:
Instead of storing `HashMap<Felt, CheckpointStorage>` for the `ValidatorStorage`, maybe we should store directly `Box<dyn FetchFromStorage>` so we don't call `build` every iterations?

Reply:
Will do that in an upcoming PR, once we have a better idea of the complete flow and data needed